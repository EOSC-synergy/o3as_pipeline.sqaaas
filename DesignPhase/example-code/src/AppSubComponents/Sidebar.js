import * as React from 'react';
import {ApprearenceSection, FilterDataSection, ModelsSection, ReferenceYearSection} from './AppSubComponents/SectionComponents';
import {Button} from './BasicComponents'

function downloadHandler() {
    return "TODO";
}


export default function Sidebar(props) {
    return (<div> {props.open && 
    (<div>
        <ApprearenceSection />
        <ModelsSection />
        <FilterDataSection />
        <ReferenceYearSection />
        <Button text="Download" onClick={downloadHandler}/>
    </div>)
    }</div>);
}