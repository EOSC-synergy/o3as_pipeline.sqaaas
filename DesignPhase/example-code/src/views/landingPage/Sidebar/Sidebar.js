import * as React from 'react';
import Section from './Section/Section.js';
import defaultStructure from '../../../config/defaultConfig.json';
import DownloadModal from './DownloadModal/DownloadModal.js';
import { setCurrentType } from '../../../store/plotSlice.js';
import { useDispatch } from "react-redux";
import PlotTypeSelector from './InputComponents/PlotTypeSelector/PlotTypeSelector.js';

/**
 * Contains all input components responsible for the modification 
 * of the plot settings.
 * @param {Object} props 
 *  props.open ->  whether sideBar should be open
 *  props.onClose -> handles closing of the sidebar
 *  props.error -> enables component to report an error
 * @returns {JSX} a jsx containing a sidebar with sections containing input components, a download button and a plotType dropdown
 */
function Sidebar(props) {

    const dispatch = useDispatch()
    let i = props.onClose;

    const [openDownloadModal, setOpenDownloadModal] = React.useState(false);
    const [expandedSection, setExpandedSection] = React.useState(null); // -> idx of sections array


    const [plotType, setPlotType] = React.useState("");

    /**
     * Changes the plot type and updates the sidebar according to new plot type
     * @param {String} newPlotType 
     */
    const changePlotType = (newPlotType) => {
        // check if newPlotType is valid
        // update plotType
        setPlotType(newPlotType);
        dispatch(setCurrentType(newPlotType))
        // rerender sections -> lose current value?
    }

    /**
     * closes the download modal
     */
    const onCloseDownloadModal = () => {
        setOpenDownloadModal(false);
    }

    /**
     * shows the download modal
     */
    const onOpenDownloadModal = () => {
        setOpenDownloadModal(true);
    }

    /**
     * collapses all sections
     */
    const onCollapseSection = () => {
        setExpandedSection(null);
    }

    /**
     * expands section with id (index in section array) i
     * collapses all other currently expanded sections
     * @param {int} i 
     */
    const onExpandSection = (i) => {
        setExpandedSection(i);
    }

    return (
    <>
        {props.open && 
            <>

                <PlotTypeSelector plotType={plotType} />

                {defaultStructure.forEach((s, idx) =>
                    <Section
                        name={defaultStructure.sections[s].name}
                        open={expandedSection === idx}
                        components={"Empty for now"}
                        onCollapse={onCollapseSection}
                        onExpand={onExpandSection}
                    />
                )}

                {/* Maybe into their own component? */}
                {/* <Button text="Download" onClick={onOpenDownloadModal}/> */}
                <DownloadModal open={openDownloadModal} onClose={onCloseDownloadModal} />
            </>
        }
    </>);
}

export default Sidebar;