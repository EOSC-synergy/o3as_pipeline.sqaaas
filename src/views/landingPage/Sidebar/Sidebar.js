import * as React from 'react';
import { styled, useTheme } from '@mui/material/styles';
import Section from './Section/Section.js';
import defaultStructure from '../../../config/defaultConfig.json';
import DownloadModal from './DownloadModal/DownloadModal.js';
import { setCurrentType } from '../../../store/plotSlice/plotSlice';
import { useDispatch } from "react-redux";
import PlotTypeSelector from './InputComponents/PlotTypeSelector/PlotTypeSelector.js';
import SwipeableDrawer from '@mui/material/SwipeableDrawer';
import { Button } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import IconButton from '@mui/material/IconButton';

/**
 * Defining a drawerheader section at the beginning of a drawer
 */
const DrawerHeader = styled('div')(({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end',
  }));

/**
 * Contains all input components responsible for the modification 
 * of the plot settings.
 * @param {Object} props 
 * @param {boolean} props.isOpen -  whether sideBar should be open
 * @param {function} props.onClose - handles closing of the sidebar
 * @param {function} props.reportError - enables component to report an error
 * @returns {JSX} a jsx containing a sidebar with sections containing input components, a download button and a plotType dropdown
 */
function Sidebar(props) {

    const theme = useTheme();

    // const dispatch = useDispatch()

    const [isDownloadModalVisible, setDownloadModalVisible] = React.useState(false);
    const [expandedSection, setExpandedSection] = React.useState(null); // -> idx of sections array

    /**
     * closes the download modal
     */
    const closeDownloadModal = () => {
        setDownloadModalVisible(false);
    }

    /**
     * shows the download modal
     */
    const openDownloadModal = () => {
        setDownloadModalVisible(true);
    }

    /**
     * collapses all sections
     */
    const collapseSection = () => {
        setExpandedSection(null);
    }

    /**
     * expands section with id (index in section array) i
     * collapses all other currently expanded sections
     * @param {int} i the section that should be expanded
     */
    const expandSection = (i) => {
        setExpandedSection(i);
    }

    return (
            <SwipeableDrawer
                anchor="right"
                open={props.isOpen}
                onClose={props.onClose}
                onOpen={props.onOpen}
                variant="persistent"
                sx= {{
                    width: 400,
                    maxWidth: "100vw",
                    flexShrink: 0,
                    '& .MuiDrawer-paper': {
                        width: 400,
                        maxWidth: "100vw",
                    },
                }}
                data-testid="sidebar"
            >
                <DrawerHeader>
                    <IconButton
                        onClick={props.onClose}
                        data-testid="sidebarClose"
                    >
                        <CloseIcon />
                    </IconButton>
                </DrawerHeader>
                    <PlotTypeSelector />

                    {defaultStructure["sections"].map((s, idx) =>
                        <Section
                            name={s.name}
                            key={idx}
                            open={expandedSection === idx}
                            components={s.components}
                            onCollapse={collapseSection}
                            onExpand={expandSection}
                        />
                    )}

                    <Button sx={{marginLeft: "10%", marginTop: "1em", width: "80%"}} variant="outlined" onClick={openDownloadModal}>Download</Button>
                    <DownloadModal isOpen={isDownloadModalVisible} onClose={closeDownloadModal} />
        </SwipeableDrawer>
    );
}

export default Sidebar;