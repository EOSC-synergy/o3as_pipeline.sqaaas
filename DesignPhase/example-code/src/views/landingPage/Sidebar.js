import * as React from 'react';
import {AppearenceSection, FilterDataSection, ModelsSection, ReferenceYearSection} from './AppSubComponents/SectionComponents';
import {Button} from './BasicComponents'

function downloadHandler() {
    return "TODO";
}


/**
 * Contains all components
 * responsible for the modification 
 * of the plot settings.
 */
export default function Sidebar(props) {
    return (<> {props.open && 
    (<div>
        <AppearenceSection />
        <ModelsSection />
        <FilterDataSection />
        <ReferenceYearSection />
        <Button text="Download" onClick={downloadHandler}/>
    </div>)
    } </>);
}