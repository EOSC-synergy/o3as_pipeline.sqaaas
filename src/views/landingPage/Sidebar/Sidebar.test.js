import React from 'react';
import ReactDOM from 'react-dom';
import Sidebar from './Sidebar';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';


it('Sidebar renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Sidebar />, div);
});

// Snapshot test
it('Sidebar renders correctly', () => {
    const treeClosed = renderer
        .create(<Sidebar isOpen={false} />)
        .toJSON();
    expect(treeClosed).toMatchSnapshot();

    const div = document.createElement('div');
    ReactDOM.render(<Sidebar isOpen={true} />, div);
    expect(div).toMatchSnapshot();
});

// expect there to be a download button
it('Sidebar has a download button', () => {
    render(<Sidebar />);
    const download = screen.getByText(/Download/i);
    expect(download).not.toBeDisabled();
})