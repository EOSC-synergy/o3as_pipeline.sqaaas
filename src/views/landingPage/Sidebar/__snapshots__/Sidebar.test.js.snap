// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked css-19iwfmc-MuiDrawer-docked"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight css-18sg6k4-MuiPaper-root-MuiDrawer-paper"
    style={
      Object {
        "pointerEvents": "",
        "visibility": "hidden",
      }
    }
  >
    <div
      className="css-cfiwj7"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
    PlotTypeSelector
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
      data-testid="section"
    >
      <div
        aria-controls="panel1a-content"
        aria-expanded={false}
        className="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
        id="panel1a-header"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            APPEARANCE
          </p>
        </div>
        <div
          className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              className="MuiAccordion-region"
              id="panel1a-content"
              role="region"
            >
              <div
                className="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
              >
                PlotNameField
                XAxisSlider
                YAxisSlider
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
      data-testid="section"
    >
      <div
        aria-controls="panel1a-content"
        aria-expanded={false}
        className="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
        id="panel1a-header"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            MODELS
          </p>
        </div>
        <div
          className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              className="MuiAccordion-region"
              id="panel1a-content"
              role="region"
            >
              <div
                className="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
              >
                ModelGroupConfigurator
                AddModelGroupModal
                EditModelGroupModal
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
      data-testid="section"
    >
      <div
        aria-controls="panel1a-content"
        aria-expanded={false}
        className="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
        id="panel1a-header"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            FILTER DATA
          </p>
        </div>
        <div
          className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              className="MuiAccordion-region"
              id="panel1a-content"
              role="region"
            >
              <div
                className="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
              >
                LatitudeBandSelector
                TimeCheckBoxGroup
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
      data-testid="section"
    >
      <div
        aria-controls="panel1a-content"
        aria-expanded={false}
        className="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
        id="panel1a-header"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            REFERENCE YEAR
          </p>
        </div>
        <div
          className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              className="MuiAccordion-region"
              id="panel1a-content"
              role="region"
            >
              <div
                className="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
              >
                ReferenceModelSelector
                ReferenceYearSlider
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Download
    </button>
  </div>
</div>
`;

exports[`Sidebar renders correctly 2`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked css-19iwfmc-MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight css-18sg6k4-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        class="css-cfiwj7"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      PlotTypeSelector
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
        data-testid="section"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              APPEARANCE
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                class="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
                >
                  PlotNameField
                  XAxisSlider
                  YAxisSlider
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
        data-testid="section"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              MODELS
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                class="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
                >
                  ModelGroupConfigurator
                  AddModelGroupModal
                  EditModelGroupModal
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
        data-testid="section"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              FILTER DATA
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                class="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
                >
                  LatitudeBandSelector
                  TimeCheckBoxGroup
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1dm7ii1-MuiPaper-root-MuiAccordion-root"
        data-testid="section"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root css-1icm36a-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              REFERENCE YEAR
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                class="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-ta84vm-MuiAccordionDetails-root"
                >
                  ReferenceModelSelector
                  ReferenceYearSlider
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Download
      </button>
    </div>
  </div>
</div>
`;
